packages/models-library/tests/test_models_fit_schemas.py

```diff
@@ -0,0 +1,52 @@
+# pylint:disable=unused-variable
+# pylint:disable=unused-argument
+# pylint:disable=redefined-outer-name
+# pylint:disable=protected-access
+import json
+from typing import Callable
+
+import pytest
+from pydantic.main import BaseModel
+
+from models_library.projects import Project
+from models_library.services import ServiceDockerData
+
+
+@pytest.mark.parametrize(
+    "pydantic_model, original_json_schema",
+    [(ServiceDockerData, "node-meta-v0.0.1.json"), (Project, "project-v0.0.1.json")],
+)
+def test_generated_schema_same_as_original(
+    pydantic_model: BaseModel,
+    original_json_schema: str,
+    diff_json_schemas: Callable,
+    json_schema_dict: Callable,
+):
+    generated_schema = json.loads(pydantic_model.schema_json(indent=2))
+    original_schema = json_schema_dict(original_json_schema)
+
+    # run one direction original schema encompass generated one
+    process_completion = diff_json_schemas(original_schema, generated_schema)
```

Comment:
MINOR: Some tips about the design. For fixtures that produce callable, we should follow a design similar to the ``mocker`` pytest plugin. Running the fixture produces a result object that incorporate meaningful assert checks

```python

mock.assert_called_once()

comparison = diff_json_schemas(...)

comparison.assert_same_name()
comparison.assert_same_schemas()
```

It is not only more readable and structure but allows fine-grain the assertion checks without having to expose the implementation details to the tester (eg. tha tit is run with a subprocess etc)

Reply:
that is true. for now it is only used there. I will do it if I need it elsewhere